crates/solver/src/settlement.rs

```diff
@@ -299,6 +318,77 @@ impl Settlement {
         }
     }
 
+    // Checks whether the settlement prices do not deviate more than max_settlement_price_deviation from the auction prices on certain pairs.
+    pub fn satisfies_price_checks(
+        &self,
+        auction_id: u64,
+        solver_name: &str,
+        external_prices: &ExternalPrices,
+        max_settlement_price_deviation: &Ratio<BigInt>,
+        tokens_to_satisfy_price_test: &TokensForPriceCheck,
```

Comment:
These two things are always being passed around together to other functions. I think they could be their own struct. Having a dedicated struct for this might even help make these names shorter. 

Suggestion. 

Struct PriceCheck with fields

Max_deviation (currently your max settlement price deviation)
Tokens (currently your tokens to satisfy price test)